### PR TITLE
refactor session list views

### PR DESCRIPTION
**Pull request by leafty**
Refactor components used to show a list of session:
* Added `AnonymousSessionsList` for anonymous sessions in `/sessions`
* Added `ProjectSessionsList` for the list of sessions shown in `/projects/<namespace>/<project>/sessions`

/deploy #persist #cypress extra-values=core.horizontalPodAutoscaling.minReplicas=1,core.replicaCount=1

### CODE REVIEW

**Review by lorenzo-cavazzi — CHANGES_REQUESTED** [review]
The code quality is really high! I added inline just an alternative solution for the sessions Cypress tests, and a generic comment on when to use `cx` (feel free to disagree on that :grin: ).

Generally, I believe we should keep suggesting how to start a session in the new Session tab when no sessions are running. I would add back an InfoAlert similar to the previous one, suggesting using the button on top (ofc no need to recreate the old button since now it's in the normal header) or clicking straight on a "New session" link that brings the user to the "Start with options" page.

image


At first sight, I'm tempted to say that duplicating the session button on the sessions tab might be redundant. Still, removing it might be confusing for some users and I can't think of a better alternative. We can live with that

image

File: tests/cypress/support/sessions/gui_commands.ts [hunk 1]
```diff
@@ -17,10 +17,12 @@
...
-  cy.get_cy("session-log-button").click();
+  cy.get(".sessionsButton").first().find("[data-cy='more-menu']").click();
+  cy.get_cy("session-log-button").filter(":visible").click();
 });
 
 Cypress.Commands.add("gui_open_session", () => {
-  cy.get_cy("open-session").click();
+  // eslint-disable-next-line cypress/no-unnecessary-waiting
+  cy.wait(1000, { log: false });
+  cy.get_cy("open-session").first().click();
```
[the remark this answers is not in the document]
Comment (leafty) [reply tests/cypress/support/sessions/gui_commands.ts]: Done in 9b84b1d.

File: client/src/features/session/components/SessionsList.tsx [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+        <span className={cx("me-3", "mt-2")}>{icon}</span>
+        <div
+          className={cx(
+            "d-flex",
+            "flex-column",
+            "align-items-start",
+            "overflow-hidden"
+          )}
+        >
+          <div className={cx("project", "d-inline-block", "text-truncate")}>
```
[the remark this answers is not in the document]
Comment (leafty) [reply client/src/features/session/components/SessionsList.tsx:229]: I have found that it makes changing and experimenting with classes faster (can add or comment one class name).

**Comment by leafty** [comment]
> Generally, I believe we should keep suggesting how to start a session in the new Session tab when no sessions are running. I would add back an InfoAlert similar to the previous one, suggesting using the button on top (ofc no need to recreate the old button since now it's in the normal header) or clicking straight on a "New session" link that brings the user to the "Start with options" page.

Yes, I will add the alert and button in another PR, which would make it easier to discuss the UI/UX for the updated sessions tab.

**Review by lorenzo-cavazzi — APPROVED** [review]
Lgtm :rocket:

**Pull request merged**